SonarQube.TeamBuild.PreProcessor/ProcessedArgs.cs

```diff
@@ -70,8 +71,31 @@ public ProcessedArgs(string key, string name, string version, bool installLoader
         /// </summary>
         public bool InstallLoaderTargets { get; private set; }
 
+        /// <summary>
+        /// Returns the combined command line and file analysis settings
+        /// </summary>
         public IAnalysisPropertyProvider AggregateProperties { get { return this.aggProperties; } }
 
+        public IAnalysisPropertyProvider LocalProperties { get { return this.cmdLineProperties; } }
+
+        /// <summary>
+        /// Returns the name of property settings file or null if there is not one
+        /// </summary>
+        public string PropertiesFileName
+        {
+            get
+            {
+                FilePropertyProvider fileProvider = this.globalFileProperties as FilePropertyProvider;
+                if (fileProvider != null)
+                {
+                    Debug.Assert(fileProvider.PropertiesFile != null, "File properties should not be null");
+                    Debug.Assert(!string.IsNullOrWhiteSpace(fileProvider.PropertiesFile.FilePath), "Settings file name should not be null");
+                    return fileProvider.PropertiesFile.FilePath;
+                }
+                return null;
+            }
+        }
+
```

Comment:
Down-casting from the interface is an ugly solution (and triggers a warning from a new SonarLint rule :-) ). This could be changed as part of refactoring the FilePropertyProvider.